payload/usr/local/sal/checkin_modules/machine_checkin.py

```diff
@@ -98,17 +100,29 @@ def get_machine_name(net_config, nametype):
     ).strip()
 
 
-def get_friendly_model(serial):
+def get_friendly_model(serial, udid):
     """Return friendly model name"""
-    if not MODEL_PATH.exists():
+
+    # set up cache file for this udid...create dir,
+    MODEL_PATH.mkdir(mode=0o755, parents=True, exist_ok=True)
+
+    # name cache for this udid
+    UDID_CACHE_PATH = pathlib.Path(MODEL_PATH, '%s.txt' % (udid))
+    for cache_file in MODEL_PATH.iterdir():
+        # clean up any other files in dir
+        if cache_file != UDID_CACHE_PATH:
+            cache_file.unlink()
```

Comment:
We should probably handle failing to remove the file for whatever reason

Reply:
Good point, no reason to kill the whole thing for something that is just housekeeping.  I added

```
# clean up any other files in dir
        if cache_file != UDID_CACHE_PATH:
            try:
                cache_file.unlink()
            except:
                pass
```

Reply:
Just wanted to follow up and see if you think this is a good way to handle failure to remove that file.